core-plugins/src/main/java/io/cdap/plugin/batch/joiner/Joiner.java

```diff
@@ -215,31 +259,57 @@ public StructuredRecord merge(StructuredRecord joinKey, Iterable<JoinElement<Str
   }
 
   FailureCollector init(Map<String, Schema> inputSchemas, FailureCollector failureCollector) {
```

Comment:
nit: Mark this as @VisibleForTesting

Reply:
looks like inputSchemasAvailable is called at many places. can we centralize that check in this method?

Reply:
Added @VisibleForTesting annotation.

inputSchemasAvailable is called from:
1. configurePipeline - part of check to verify that output schema is provided if input schemas are not known
2. prepareRun - don't record lineage if input schemas are not available
3. createOutputFieldInfos - bail out if input schemas are not available
4. JoinerConfig.validateJoinKeySchemas - part of check to skip validation if input schemas are not available or joinKeys is a macro. 
Since it's used in many different contexts, I don't think it makes sense to centralize the check in init().